src/socketSetup.ts

```diff
@@ -0,0 +1,78 @@
+import { Server } from 'socket.io';
+import { chatRepository } from './repositories/chat.repository';
+import { wordChecker } from './application/utils/wordChecker/wordChecker';
+// import { userRepository } from './repositories/user.repository';
+
+export const setupSocket = (io: Server) => {
+  io.on('connection', (socket) => {
+    console.log('a user connected');
+
+    socket.on('join', async (chatId) => {
+      socket.join(chatId);
+      console.log(`User joined chat ${chatId}`);
+      try {
+        const chat = await chatRepository.getById(chatId);
+
+        chat.messages.forEach((message) => {
+          socket.emit('chat message', `${message.message}`);
+        });
+      } catch (error) {
+        console.error('Error getting chat entity:', error);
+      }
+    });
+
+    socket.on('leave', (chatId) => {
+      socket.leave(chatId);
+      console.log(`User left chat ${chatId}`);
+    });
+
+    socket.on('chat message', async (msg, chatId) => {
+      console.log(`message: ${msg} in chat ${chatId}`);
+
+      try {
+        const chat = await chatRepository.getById(chatId);
+        // const user = await userRepository.getById(
+        //   '0d81332b4945a796c42abb425e000e66',
+        // );
+        const msgWithoutjunk = msg.replace(/[^a-zA-Z\s]/g, '');
```

Comment:
In socket.on('chat message'), check for profanity or banned words before allowing messages to be broadcast.


  ```typescript
  try {
    if (containsBannedWords(msg)) {
      // Handle the case where a banned word is found
      socket.emit('banned word detected', 'Your message contains inappropriate content.');
      return; // Stop processing the message
    }
    ```

Reply:
Thank you for reviewing! Added badwords library check in socket with my last commit.

Reply:
Thank you :) this change I believe more like for the future version of the application, but still there is a reason to start taking caring of that.